tensorflow/compiler/xla/service/gpu/ir_emission_utils.cc

```diff
@@ -233,12 +233,18 @@ llvm::Value* EmitFullWarpShuffleDown(llvm::Value* value, llvm::Value* offset,
 
   // Special case for efficiency
   if (value->getType()->isFloatTy() && bit_width == 32) {
-    return EmitDeviceFunctionCall(
-        "amdgcn_shfl_down",
-        {value, offset},
-        {F32, S32}, F32, {}, builder, module);
+
+    llvm::Value* value_as_int = builder->CreateBitCast(value, builder->getIntNTy(bit_width));
+    llvm::Value*  result = EmitDeviceFunctionCall(
+        "__ockl_readuplane_i32",
```

Comment:
@sumesh13 logic here is AMD-only. is it possible to also encapsulate this into LLVMTargetFeature or TargetIRBuilder?

Reply:
Yeah ...will do that when I bring in my upstream changes locally